pkg/instrumentation/apachehttpd.go

```diff
@@ -0,0 +1,236 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instrumentation
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+)
+
+const (
+	apacheConfigDirectory         = "/usr/local/apache2/conf"
+	apacheConfigFile              = "httpd.conf"
+	apacheAgentConfigFile         = "opentemetry_agent.conf"
+	apacheAgentDirectory          = "/opt/opentelemetry-webserver"
+	apacheAgentSubDirectory       = "/agent"
+	apacheAgentDirFull            = apacheAgentDirectory + apacheAgentSubDirectory
+	apacheAgentConfigDirectory    = "/source-conf"
+	apacheAgentConfDirFull        = apacheAgentDirectory + apacheAgentConfigDirectory
+	apacheAgentInitContainerName  = "otel-agent-attach-apache"
+	apacheAgentCloneContainerName = "otel-agent-source-container-clone"
+	apacheAgentConfigVolume       = "otel-apache-conf-dir"
+	apacheAgentVolume             = "otel-apache-agent"
+	apacheAttributesEnvVar        = "OTEL_APACHE_AGENT_CONF"
+	apacheServiceInstanceId       = "<<SID-PLACEHOLDER>>"
+	apacheServiceInstanceIdEnvVar = "APACHE_SERVICE_INSTANCE_ID"
+)
+
+func injectApacheHttpdagent(_ logr.Logger, apacheSpec v1alpha1.ApacheHttpd, pod corev1.Pod, index int, otlpEndpoint string, resourceMap map[string]string) corev1.Pod {
```

Comment:
can you add a comment here about what exactly this method is doing? i'm a bit confused by the two `isApacheInitContainerMissing` checks.

Reply:
I agree - it's worth explaining, since for Apache HTTPD, it works a bit specifically. Comments included in the code.